### ISSUE TITLE
How to improve the GPU usage?

### ISSUE THREAD

**Issue by celsofranssa** [MaartenGr/KeyBERT#198]
I am extracting keywords from 2.5 million sentences, iterating through the list of sentences. However, this approach is only using 8% of the GPU.

Would there be a more efficient approach?

**Comment by lfoppiano** [comment]
@celsofranssa Yes, you can pass a list of sentences at each call to keybert, and then retrieve the keywords for each sentence in the same index position.

**Comment by MaartenGr** [comment]
@celsofranssa It is difficult to say what is happening here. 8% is tricky to interpret without knowing the full context. For instance, does that relate to the embedding step of the process (which you can perform manually beforehand)? Also, it depends on the underlying embedding model that you use. Without seeing the code, it is difficult to say.
